main.go

```diff
@@ -2,27 +2,45 @@ package main
 
 import (
 	"crypto/tls"
+	"crypto/x509"
 	"flag"
 	"fmt"
+	"time"
 )
 
-func testTLS(host string) (valid bool, reason string) {
-	conn, err := tls.Dial("tcp", host, &tls.Config{})
+func getTLS(host string) (valid bool, cert []*x509.Certificate, reason string) {
+	conn, err := tls.Dial("tcp", host, &tls.Config{InsecureSkipVerify: true})
 	if err != nil {
-		return false, err.Error()
+		cert = conn.ConnectionState().PeerCertificates
+		return false, cert, err.Error()
 	}
 	conn.Close()
-	return true, ""
+	cert = conn.ConnectionState().PeerCertificates
+	return true, cert, ""
+}
+
+func testExpiration(i time.Time) bool {
+	if time.Now().After(i) {
+		return false
+	}
+	return true
 }
 
 func main() {
 	flag.Parse()
 	for _, v := range flag.Args() {
-		ok, msg := testTLS(v)
+		ok, certs, msg := getTLS(v)
+		var cert = &x509.Certificate{}
+		cert = certs[0]
 		if !ok {
 			fmt.Printf("%s: notok: %s\n", v, msg)
 		} else {
-			fmt.Printf("%s: ok\n", v)
+			foo := testExpiration(cert.NotAfter)
```

Comment:
What do you think about a better variable name for `foor`. Maybe `certficateExpired`?